airbyte-integrations/connectors/source-oracle-v2/src/cdk/main/kotlin/io/airbyte/cdk/operation/Operation.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.operation
+
+const val CONNECTOR_OPERATION: String = "airbyte.connector.operation"
+
+/**
+ * Interface that defines a CLI operation. Each operation maps to one of the available
+ * [OperationType]s.
+ */
+interface Operation {
+
+    val type: OperationType
+
+    fun execute()
+}
+
+/**
+ * Defines the operations that may be invoked via the CLI arguments. Not all connectors will
+ * implement all of these operations.
+ */
+enum class OperationType {
```

Comment:
again, I feel like this is where we could/should use sealed classes. Also, I don't like the fact that all operations are centralized, when they'll always be split between source and destination (except for the spec)

Reply:
I might be missing something - why do we need this enum at all? e.g.
* `ConnectorCommandLinePropertySource` accepts a list<string> of operations instead of calling `OperationType.entries`
* then we can do `class DestinationConnectorCommandLinePropertySource: ConnectorCommandLinePropertySource(listOf("spec", "check", "write"))`
    * admittedly, I don't know how much of a PITA it is to inject this thing so 🤷 

(we could define a DestinationOperations enum if we wanted, I guess. But afaict we never reference the operation enun values except to get their name)

Reply:
> we never reference the operation enun values except to get their name
that's true, but IMO, that's also OK

> I don't like the fact that all operations are centralized
only this enum is centralized, and it's set of values is a quasi-constant

I'll try to get rid of it, I like @edgao 's idea